src/main/java/io/fabric8/maven/docker/BuildMojo.java

```diff
@@ -34,7 +35,11 @@ protected void executeInternal(ServiceHub hub) throws DockerAccessException, Moj
         if (skipBuild) {
             return;
         }
-        for (ImageConfiguration imageConfig : getResolvedImages()) {
+        List<ImageConfiguration> resolvedImages = getResolvedImages();
+        if(resolvedImages == null || resolvedImages.isEmpty()) {
```

Comment:
resolvedImges can't be null otherwise the next for loop would fail with an NPE.

Actually I would either put this into an extra method which returns early in this case or put this into an if-else with the following loop.